src/org/traccar/protocol/EelinkProtocolEncoder.java

```diff
@@ -54,6 +54,17 @@ protected Object encodeCommand(Command command) {
                 return encodeContent("RELAY,0#");
             case Command.TYPE_REBOOT_DEVICE:
                 return encodeContent("RESET#");
+            case Command.TYPE_GET_VERSION:
+                return encodeContent("VERSION#");
+            case Command.TYPE_GET_DEVICE_STATUS:
+                return encodeContent("STATUS#");
+            case Command.TYPE_SOS_NUMBER:
+                String num = command.getString(Command.KEY_DATA);
+                if (!num.isEmpty()) {
```

Comment:
Please don't reply to every comment. Fix everything that you can/agree, and then we can discuss other ones.